src/NBitcoin/ChainedBlock.cs

```diff
@@ -75,15 +75,14 @@ public ChainedBlock(BlockHeader header, uint256 headerHash, ChainedBlock previou
         /// </summary>
         /// <param name="header">The header for the block.</param>
         /// <param name="height">The height of the block.</param>
-        public ChainedBlock(BlockHeader header, int height)
+        public ChainedBlock(BlockHeader header, uint256 headerHash, int height)
```

Comment:
Not sure if it's good idea to provide headerHash separately instead of calling `header.GetHash(); ` inside. 
Maybe it would be better to add a constructor overload so you can keep old and the new one? Looks like new constructor is needed only for a few cases but it forces it''s usage in all cases.

If you'll do this revert all the usages of the new constructor to use old constructor and use new one only when you need it.

Reply:
The NetworkOptions required to compute the hash are more often available at a shallower call level.